packages/firestore/src/api/snapshot.ts

```diff
@@ -520,6 +520,13 @@ export class DocumentSnapshot<
       document.data.value.mapValue.fields,
       'previous'
     );
+    if (this.metadata.hasPendingWrites) {
+      throw new FirestoreError(
+        Code.FAILED_PRECONDITION,
+        'DocumentSnapshot.toJSON attempted to serialize a document with pending writes. ' +
+          'Await `waitForPendingWrites()` before invoking `toJSON()`.'
```

Comment:
I think it needs to be `'Await `waitForPendingWrites()` before getting documents.'